git/repo.go

```diff
@@ -345,3 +350,92 @@ func (repo *Repository) ObjectForPath(root Object, pathstr string) (Object, erro
 
 	return node, nil
 }
+
+// ParseCommitList executes a custom git log command of the specified branch of the
+// associated git repository and returns the resulting list of commits as an array.
+func (repo *Repository) ParseCommitList(branch string) ([]wire.CommitListItem, error) {
+	gdir := fmt.Sprintf("--git-dir=%s", repo.Path)
+
+	usefmt := "--pretty=format:"
+	usefmt += "Commit:=%H%n"
+	usefmt += "Committer:=%cn%n"
+	usefmt += "Author:=%an%n"
+	usefmt += "Date-iso:=%ai%n"
+	usefmt += "Date-rel:=%ar%n"
+	usefmt += "Subject:=%s%n"
+	usefmt += "Changes:="
+
+	cmd := exec.Command("git", gdir, "log", branch, usefmt, "--name-status")
```

Comment:
I assumed that the git log command would be optimized and therefore faster than what we can re-implement and went for it. I will leave it for now, but will create an issue pointing to this function and #20.